### PR TITLE
Added automatic pushing to Quay

### LINKED ISSUE #82: Automate timbot container build and publish to quay.io

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Fixes #82

### CODE REVIEW

File: .github/workflows/quay.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: quay
+
+on:
+  push:
+    branches:
+      - master
```
Comment (timflannagan) [inline .github/workflows/quay.yml]: We may want to just have this trigger on pull_requests/push at the moment, so we don't have to force merge to get it working initially.

File: .github/workflows/quay.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    steps:
+    - name: update container on quay.io
+      run: curl -H $header --data $payload ${{ secrets.QUAY_WEBHOOK_URL }}
```
Comment (timflannagan) [inline .github/workflows/quay.yml]: You should add `run: set -x; curl ...` so it leaks the `${{ secrets.QUAY_WEBHOOK_URL }}` in the VM.

File: .github/workflows/quay.yml (hunk 1 above)
Comment (nathan-weinberg) [reply .github/workflows/quay.yml]: Ah just for testing you mean? Then put it back when we merge?

File: .github/workflows/quay.yml (hunk 2 above)
Comment (nathan-weinberg) [reply .github/workflows/quay.yml]: Don't we not want the secret in the VM?

File: .github/workflows/quay.yml (hunk 1 above)
Comment (timflannagan) [reply .github/workflows/quay.yml]: Yeah, exactly. We can revert back once we try and play around with it a bit more.

File: .github/workflows/quay.yml (hunk 2 above)
Comment (timflannagan) [reply .github/workflows/quay.yml]: Sure, but I was just messing around with that comment. All `set -x` is doing is increased verbosity of the logs, and should, in theory, leak the value of the `{{ secrets.QUAY_WEBOOK_URL }}` unless there's logic in place in the actions that would prevent that.

File: .github/workflows/quay.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    env:
+      header: 'Content-Type:application/json'
+      payload: '{"commit":"${GITHUB_SHA}","ref":"${GITHUB_REF}","default_branch":"master"}'
+      url: '${{ secrets.QUAY_WEBHOOK_URL }}'
+    steps:
+    - name: update container on quay.io
+      run: curl -H "$header" -d "$payload "$url"
```
Comment (timflannagan) [inline .github/workflows/quay.yml]: This looks potentially problematic as you're double-quoting in the run call `-d "$payload"` but the actual payload variable contains both single and double-quotes. I would try and deflate this entire `jobs.push.env` into the run section:

```bash
run: |
    curl -H "Content-Type: application/json" -d ..."
```

Or just explicitly dump the output of those environment variables in the `run` section as well to make sure it's rendering correctly.

Comment (nathan-weinberg) [reply .github/workflows/quay.yml]: I think this was the first thing I tried, but I'll keep messing with it

**Comment by nathan-weinberg** [comment]
Closing this - work has to be done in actual repo on a working branch - see https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

**Pull request closed without merging**